cli/compose/convert/volume.go

```diff
@@ -61,17 +61,18 @@ func convertVolumeToMount(
 		return result, errors.Errorf("undefined volume %q", volume.Source)
 	}
 
+	if stackVolume.External.External && stackVolume.External.Name != "" {
+		return result, errors.Errorf("cannot specify external volume name %q", stackVolume.External.Name)
+	}
 	result.Source = namespace.Scope(volume.Source)
 	result.VolumeOptions = &mount.VolumeOptions{}
 
 	if volume.Volume != nil {
 		result.VolumeOptions.NoCopy = volume.Volume.NoCopy
 	}
 
-	// External named volumes
-	if stackVolume.External.External {
-		result.Source = stackVolume.External.Name
-		return result, nil
```

Comment:
@dnephin Thanks for your review. 
1.  I have added the block back. Could you tell me how to add a deprecation message in the loader? I am not familiar with the loader code.
2. ci/circleci: validate test failed.  It fails in some vendor Makefile, but I haven't changed anything related to this. Could you help me to take a look?  Thanks!